### ISSUE TITLE
Can only compile Admob with Unity 2019.1.13f1 (64-bit) on Windows 10

### ISSUE THREAD

**Issue by Nitrome-Steed** [googleads/googleads-mobile-unity#1113]
Admob refuses to compile a working project for me in versions of Unity available on Unity Hub.

I tried importing Unity Admob into an empty project and compiling it. This is what logcat reports:

`C:\Users\astee\AppData\Local\Android\Sdk\platform-tools>
C:\Users\astee\AppData\Local\Android\Sdk\platform-tools>adb logcat
--------- beginning of main
12-18 16:10:53.764  6854  6869 I zygote  : Background concurrent copying GC freed 195458(10MB) AllocSpace objects, 47(2MB) LOS objects, 50% free, 12MB/25MB, paused 130us total 137.171ms
12-18 16:10:58.110  3711  3973 D ClClient: Not sending keepalive.  Current connection state=STOPPED
12-18 16:11:00.077  3711  5400 D EventNotificationJob: Running EventNotificationJob, isDetail=true
12-18 16:11:00.091  3711  5400 D EventNotificationJob: processEventList() on [0] events. First cleaning up expired notifications.
12-18 16:11:00.092  3711  5400 D EventNotificationJob: Finished cleaning up notifications. Generating new notifications next
12-18 16:11:02.351   863  1076 E ANDR-PERF-MPCTL: Invalid profile no. 0, total profiles 0 only
--------- beginning of system
12-18 16:11:06.088   877   877 I SFPerfTracer:      triggers: (rate: 16:4124) (compose: 0:31) (post: 0:150) (render: 0:134) (55:496265 frames) (56:527649)
12-18 16:11:06.088   877   877 D SFPerfTracer:        layers: (4:9) (StatusBar#0 (0xad944000): 0:30641) (com.android.systemui.ImageWallpaper#0 (0xad6f9000): 0:3385) (DimLayerController/Stack=0#0 (0xad93b000): 0:833)* (Sprite#0 (0xad670000): 0:16)* (NavigationBar#0 (0xad609000): 0:795) (com.motorola.launcher3/com.android.launcher3.CustomizationPanelLauncher#0 (0xad67e000): 0:26)* (com.motorola.launcher3/com.android.launcher3.CustomizationPanelLauncher#1 (0xad148000): 56:195) (animation background stackId=1#0 (0xad17e000): 0:2)- (Application Error: com.nitrome.sillysausage#0 (0xacf0a000): 0:21)-
12-18 16:11:06.634   877  1213 I SFPerfTracer:      triggers: (rate: 4:123) (465820 sw vsyncs) (0 skipped) (62:548560 vsyncs) (64:1039403)
12-18 16:11:06.707  2649  9198 I ActivityManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=com.nitrome.sillysausage/com.unity3d.player.UnityPlayerActivity bnds=[241,806][440,1072]} from uid 10048
12-18 16:11:06.731  2649  7910 I ActivityManager: Start proc 5416:com.nitrome.sillysausage/u0a204 for activity com.nitrome.sillysausage/com.unity3d.player.UnityPlayerActivity
12-18 16:11:06.731  5416  5416 E SchedPolicy: open of /dev/cpuctl/bg_non_interactive/tasks failed: No such file or directory
12-18 16:11:04.943   863  1076 I chatty  : uid=0(root) vendor.qti.hard identical 1 line
12-18 16:11:06.630   863  1076 E ANDR-PERF-MPCTL: Invalid profile no. 0, total profiles 0 only
12-18 16:11:06.783  5416  5416 D AndroidRuntime: Shutting down VM
--------- beginning of crash
12-18 16:11:06.783  5416  5416 E AndroidRuntime: FATAL EXCEPTION: main
12-18 16:11:06.783  5416  5416 E AndroidRuntime: Process: com.nitrome.sillysausage, PID: 5416
12-18 16:11:06.783  5416  5416 E AndroidRuntime: java.lang.RuntimeException: Unable to get provider com.google.android.gms.ads.MobileAdsInitProvider: java.lang.IllegalStateException:
12-18 16:11:06.783  5416  5416 E AndroidRuntime:
12-18 16:11:06.783  5416  5416 E AndroidRuntime: ******************************************************************************
12-18 16:11:06.783  5416  5416 E AndroidRuntime: * The Google Mobile Ads SDK was initialized incorrectly. AdMob publishers    *
12-18 16:11:06.783  5416  5416 E AndroidRuntime: * should follow the instructions here:                                       *
12-18 16:11:06.783  5416  5416 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
12-18 16:11:06.783  5416  5416 E AndroidRuntime: * to add a valid App ID inside the AndroidManifest.                          *
12-18 16:11:06.783  5416  5416 E AndroidRuntime: * Google Ad Manager publishers should follow instructions here:              *
12-18 16:11:06.783  5416  5416 E AndroidRuntime: * https://googlemobileadssdk.page.link/ad-manager-android-update-manifest.   *
12-18 16:11:06.783  5416  5416 E AndroidRuntime: ******************************************************************************
12-18 16:11:06.783  5416  5416 E AndroidRuntime:
12-18 16:11:06.783  5416  5416 E AndroidRuntime:
12-18 16:11:06.783  5416  5416 E AndroidRuntime:        at android.app.ActivityThread.installProvider(ActivityThread.java:6356)
12-18 16:11:06.783  5416  5416 E AndroidRuntime:        at android.app.ActivityThread.installContentProviders(ActivityThread.java:5919)
12-18 16:11:06.783  5416  5416 E AndroidRuntime:        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5836)
12-18 16:11:06.783  5416  5416 E AndroidRuntime:        at android.app.ActivityThread.-wrap1(Unknown Source:0)
12-18 16:11:06.783  5416  5416 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1697)
12-18 16:11:06.783  5416  5416 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:106)
12-18 16:11:06.783  5416  5416 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:164)
12-18 16:11:06.783  5416  5416 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6626)
12-18 16:11:06.783  5416  5416 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
12-18 16:11:06.783  5416  5416 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
12-18 16:11:06.783  5416  5416 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:811)
12-18 16:11:06.783  5416  5416 E AndroidRuntime: Caused by: java.lang.IllegalStateException:
12-18 16:11:06.783  5416  5416 E AndroidRuntime:
12-18 16:11:06.783  5416  5416 E AndroidRuntime: ******************************************************************************
12-18 16:11:06.783  5416  5416 E AndroidRuntime: * The Google Mobile Ads SDK was initialized incorrectly. AdMob publishers    *
12-18 16:11:06.783  5416  5416 E AndroidRuntime: * should follow the instructions here:                                       *
12-18 16:11:06.783  5416  5416 E AndroidRuntime: * https://googlemobileadssdk.page.link/admob-android-update-manifest         *
12-18 16:11:06.783  5416  5416 E AndroidRuntime: * to add a valid App ID inside the AndroidManifest.                          *
12-18 16:11:06.783  5416  5416 E AndroidRuntime: * Google Ad Manager publishers should follow instructions here:              *
12-18 16:11:06.783  5416  5416 E AndroidRuntime: * https://googlemobileadssdk.page.link/ad-manager-android-update-manifest.   *
12-18 16:11:06.783  5416  5416 E AndroidRuntime: ******************************************************************************
12-18 16:11:06.783  5416  5416 E AndroidRuntime:
12-18 16:11:06.783  5416  5416 E AndroidRuntime:
12-18 16:11:06.783  5416  5416 E AndroidRuntime:        at com.google.android.gms.internal.ads.zzxi.attachInfo(com.google.android.gms:play-services-ads-lite@@18.2.0:27)
12-18 16:11:06.783  5416  5416 E AndroidRuntime:        at com.google.android.gms.ads.MobileAdsInitProvider.attachInfo(com.google.android.gms:play-services-ads-lite@@18.2.0:3)
12-18 16:11:06.783  5416  5416 E AndroidRuntime:        at android.app.ActivityThread.installProvider(ActivityThread.java:6353)
12-18 16:11:06.783  5416  5416 E AndroidRuntime:        ... 10 more
12-18 16:11:06.786  2649  9198 W ActivityManager:   Force finishing activity com.nitrome.sillysausage/com.unity3d.player.UnityPlayerActivity
12-18 16:11:06.800  2649  2673 I ActivityManager: Showing crash dialog for package com.nitrome.sillysausage u0
12-18 16:11:06.801  2649  2672 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
12-18 16:11:06.801  2649  2672 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
12-18 16:11:06.838  2649  5879 I OpenGLRenderer: Initialized EGL, version 1.4
12-18 16:11:06.838  2649  5879 D OpenGLRenderer: Swap behavior 2
12-18 16:11:07.298  2649  2672 W ActivityManager: Activity pause timeout for ActivityRecord{e256c58 u0 com.nitrome.sillysausage/com.unity3d.player.UnityPlayerActivity t152 f}
12-18 16:11:07.308  2649  2672 I ActivityManager: moveHomeStack, setupComplete:true
12-18 16:11:07.440 28188 28188 W StaticLayout: maxLineHeight should not be -1.  maxLines:2 lineCount:2
12-18 16:11:07.590  2649  2672 I ActivityManager: Killing 3693:com.meetup/u0a154 (adj 906): empty #17
12-18 16:11:07.442 28188 28188 I chatty  : uid=10078 com.google.android.googlequicksearchbox:search identical 6 lines
12-18 16:11:07.442 28188 28188 W StaticLayout: maxLineHeight should not be -1.  maxLines:2 lineCount:2
12-18 16:11:07.591  2649  2674 W zygote  : kill(-3693, 9) failed: No such process
12-18 16:11:07.603  2649  2674 I chatty  : uid=1000(system) ActivityManager identical 1 line
12-18 16:11:07.619  2649  2674 W zygote  : kill(-3693, 9) failed: No such process
12-18 16:11:07.619  2649  2674 I zygote  : Successfully killed process cgroup uid 10154 pid 3693 in 27ms
12-18 16:11:07.940  4022  4204 W OpenGLRenderer: Incorrectly called buildLayer on View: ShortcutAndWidgetContainer, destroying layer...
12-18 16:11:07.940  4022  4204 W OpenGLRenderer: Incorrectly called buildLayer on View: ShortcutAndWidgetContainer, destroying layer...
12-18 16:11:08.109  3711  3973 D ClClient: Not sending keepalive.  Current connection state=STOPPED
12-18 16:11:13.510   863  1076 E ANDR-PERF-MPCTL: Invalid profile no. 0, total profiles 0 only
12-18 16:11:13.566  2649  9198 W ActivityManager:   Force finishing activity com.nitrome.sillysausage/com.unity3d.player.UnityPlayerActivity
12-18 16:11:13.568  2649  9198 I ActivityManager: Killing 5416:com.nitrome.sillysausage/u0a204 (adj 900): crash
12-18 16:11:13.569  2649  2674 W zygote  : kill(-5416, 9) failed: No such process
12-18 16:11:13.584  2649  2674 W zygote  : kill(-5416, 9) failed: No such process
12-18 16:11:13.584  2649  2674 I zygote  : Successfully killed process cgroup uid 10204 pid 5416 in 15ms
12-18 16:11:13.797  2649  3289 W AppOps  : Finishing op nesting under-run: uid 1000 pkg android code 24 time=0 duration=0 nesting=0
12-18 16:11:16.485  3532  3609 I QCNEJ   : |CORE| CNE received action RSSI/Link Changed events: android.net.wifi.RSSI_CHANGED
12-18 16:11:16.485  3532  3609 D QCNEJ   : |CORE| Updating RSSI: -55
12-18 16:11:16.497  5362  5385 I CarrierServices: [10225] bxu.a: FiST: Wifi connectivity status true
12-18 16:11:16.498  5362  5497 I CarrierServices: [10253] awo.handleMessage: RCS is disabled, skipping config update request
12-18 16:11:16.500  5362  5496 I CarrierServices: [10252] cen.a: (PEv2-SM): WaitingForGoogleTosState: event Connectivity event
12-18 16:11:16.500  5362  5496 I CarrierServices: [10252] cdv.a: (PEv2-SM): EnabledState: event Connectivity event
12-18 16:11:16.500  5362  5385 I CarrierServices: [10225] byb.d: Connected state: [1], networkType: [WIFI]`

I have researched this error and this is the manifest that the admob library created:

`<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="com.google.unity.ads" android:versionName="1.0" android:versionCode="1">
  <application>
    <uses-library android:required="false" android:name="org.apache.http.legacy" />
    <meta-data android:name="com.google.android.gms.ads.APPLICATION_ID" android:value="ca-app-pub-################~##########" />
  </application>
</manifest>`

(I have obfuscated the application id, I also tried other application ids that are already in production. I even tried placement ids as a sanity check.)

I was recommended Unity 2019.1.13f1 and this the only version of Unity I have been able to compile the empty project with Admob in. This version of Unity requires android-ndk-r16b which I downloaded an archive of.

I have tried to trick other versions of Unity into using a different NDK as the current ones seem to all be locked into using NDK 19.0.5232133. This can be done by changing the source.properties file in the NDK folder (Unity only looks at this for confirmation). The same error occurs in all other circumstances.

I do not fully understand whether the fault lies in the Admob library, the Unity Editor, or the NDK. (This is my last day at work before Christmas break and I need to ship a build - I've done all the research I can.)

I would appreciate comments by anyone with better understanding of this anomaly.

Currently the advice on this error is to modify the manifest - something which Unity already performs correctly. This means that the error reported by the Admob library is misleading - it has already cost me an entire week of fruitless experimentation and following incorrect advice. I hope this post will save others in the same situation.

**Comment by stowy** [comment]
Hi @Nitrome-Steed 

Sorry to hear of your experience. For future reference the best place to get assistance on these issues is [our developer forum](https://groups.google.com/forum/#!categories/google-admob-ads-sdk/game-engines). I would recommend starting a thread there and include reproducing steps and ideally a sample project.

I have built Unity projects containing the GMA SDK with Unity version 2019.1.14f1 and 2019.2.9f1 without seeing the error you're describing, so I suspect that there may be something else at play here. A sample project would be a great help in reproducing the issue.

**Comment by Nitrome-Steed** [comment]
I did not mention 2019.1.14 or 2019.2.9f1. Neither of those are available through Unity Hub at this time. I have currently been reduced to trying whatever random version people recommend and trying to download it - which takes me hours of downloading. I have been doing nothing else but constantly downloading versions of Unity and NDKs for weeks.

I'll try to create a sample project for the developer forum.

**Comment by stowy** [comment]
Hi @Nitrome-Steed 

I do not believe this is a plugin issue, more of a project set up issue. It should be able to be resolved using whatever version of Unity you are using, but I am unable to assist without a sample project.  The [developer forum](https://groups.google.com/forum/#!categories/google-admob-ads-sdk/game-engines) would be a more appropriate place to get expedited assistance.

**Comment by Nitrome-Steed** [comment]
I discovered an issue with Unity 2020.1.0a16 and Admob after trying to get the plugin to work in an empty project. It may be related to the Unity package Timeline (version 1.2) - which throws errors when the project is downgraded to 2019 (the Timeline package needs to be rolled back to 1.1 and the Library folder must be deleted). Once the project is downgraded to 2019.2.17f it will compile with nothing in it and just Admob.

I tried to join Google Groups and post this but every time I try to post the thread never appears.

I can't get assistance from the developer forum.
